File: web/src/apis/domain/services/AiSessionController.ts
import type { Executor } from '../'
import type { AiSessionDto } from '../model/dto/'
import type { AiSessionInput } from '../model/static/'

export class AiSessionController {
  constructor(private executor: Executor) {}

  /**
   * 批量删除会话
   * @param ids 会话id列表
   */
  delete: (options: AiSessionControllerOptions['delete']) => Promise<void> = async (options) => {
    let _uri = '/session'
    return (await this.executor({
      uri: _uri,
      method: 'DELETE',
      body: options.body
    })) as Promise<void>
  }

  /**
   * 根据id查询会话
   * @param id 会话id
   * @return 会话信息
   */
  findById: (options: AiSessionControllerOptions['findById']) => Promise<AiSessionDto> = async (
    options
  ) => {
    let _uri = '/session/'
    _uri += encodeURIComponent(options.id)
    return (await this.executor({ uri: _uri, method: 'GET' })) as Promise<AiSessionDto>
  }

  /**
   * 查询当前登录用户的会话
   *
   * @return 会话列表
   */
  findByUser: () => Promise<Array<AiSessionDto>> = async () => {
    let _uri = '/session/user'
    return (await this.executor({ uri: _uri, method: 'GET' })) as Promise<Array<AiSessionDto>>
  }

  /**
   * 保存会话
   * @param input 会话dto参考src/main/dto/AiSession.dto
   * @return 创建后的id
   */
  save: (options: AiSessionControllerOptions['save']) => Promise<string> = async (options) => {
    let _uri = '/session/save'
    return (await this.executor({
      uri: _uri,
      method: 'POST',
      body: options.body
    })) as Promise<string>
  }
}

export type AiSessionControllerOptions = {
  findById: {
    /**
     * 会话id
     */
    id: string
  }
  save: {
    /**
     * 会话dto参考src/main/dto/AiSession.dto
     */
    body: AiSessionInput
  }
  findByUser: {}
  delete: {
    /**
     * 会话id列表
     */
    body: Array<string>
  }
}

File: src/main/java/org/integration/ai/controller/AiSessionController.java
package org.integration.ai.controller;


import lombok.AllArgsConstructor;
import org.integration.ai.domain.R;
import org.integration.ai.domain.session.dto.AiSessionInput;
import org.integration.ai.domain.session.entity.AiSession;
import org.integration.ai.services.session.AiSessionServiceImpl;
import org.integration.ai.services.session.IAiSessionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import org.springframework.stereotype.Controller;

import java.util.List;

/**
 * <p>
 *  前端控制器
 * </p>
 *
 * @author yangsen
 * @since 2025-04-21
 */
@RequestMapping("session")
@RestController
@AllArgsConstructor
public class AiSessionController {

    @Autowired
    public AiSessionServiceImpl aiSessionService;

    /**
     * 根据id查询会话
     * @param id 会话id
     * @return 会话信息
     */
    @GetMapping("{id}")
    public R<AiSession> findById(@PathVariable String id) {
        return R.ok(aiSessionService.findById(id));
    }

    /**
     * 保存会话
     * @param input AiSessionInput
     * @return 创建后的id
     */
    @PostMapping("save")
    public R<String> save(@RequestBody AiSessionInput input) {
        return R.ok(aiSessionService.save(input).getId());
    }

    /**
     * 查询当前登录用户的会话
     *
     * @return 会话列表
     */
    @GetMapping("user")
    public R<List<AiSession>> findByUser() {
        return R.ok(aiSessionService.findByUser());
    }

    /**
     * 批量删除会话
     * @param ids 会话id列表
     */
    @DeleteMapping
    public R<Integer> delete(@RequestBody List<String> ids) {
        return R.ok(aiSessionService.delete(ids));
    }
}
